cursor.go

```diff
@@ -233,11 +218,14 @@ func (c *Cursor) nextLocked(dest interface{}) (bool, error) {
 			} else {
 				c.buffer = append(c.buffer, value)
 			}
+			c.applyPendingSkips(true)
```

Comment:
Could you explain why this call is needed here since we also call `applyPendingSkips` in `seekCursor`. I assume it is due to the `atomConverted` flag, if so could you add some more info the the comment for `applyPendingSkips` explaining what it does?